### PR TITLE
use reliable and ordered packets for avatar identity

### PR DESCRIPTION
**Pull request by sethalves**
- use reliable and ordered packets for avatar identity

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 96174468b3b3ac36d187188cca4f52086a028217 available at: https://gist.github.com/db4768aebd6a2d5abc587f03e88f73ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7824.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7824.dmg

**Comment by sethalves** [comment]
to test:
- mix and match multiple avatars in old (pre this PR) and new (running this code) domains using old and new interfaces
- make sure everyone can see each other and movements look okay

File: libraries/avatars/src/AvatarData.cpp [hunk 1]
```diff
@@ -1193,10 +1193,15 @@ void AvatarData::sendIdentityPacket() {
 
     QByteArray identityData = identityByteArray();
 
-    auto identityPacket = NLPacket::create(PacketType::AvatarIdentity, identityData.size());
-    identityPacket->write(identityData);
-
-    nodeList->broadcastToNodes(std::move(identityPacket), NodeSet() << NodeType::AvatarMixer);
+    auto packetList = NLPacketList::create(PacketType::AvatarIdentity, QByteArray(), true, true);
```
Comment (Atlante45) [inline libraries/avatars/src/AvatarData.cpp]: Interesting, does the AvatarIdentity packet not fit in a single packet anymore?

Comment (Atlante45) [inline libraries/avatars/src/AvatarData.cpp]: Oh I guess that way we can handle big URLs.

Comment (sethalves) [inline libraries/avatars/src/AvatarData.cpp]: It generally does, but I'm sneaking up on bigger changes.  This is groundwork.

**Comment by kevinmthomas-carpool** [comment]
@sethalves are you looking to have a single user load an avatar on one computer, with a pre-PR build and then load the same avatar on the PR build on another computer at the same time and ensure each can see each other?

**Comment by sethalves** [comment]
I'm basically wanting to make sure that clients from before this PR can connect to servers from this PR and that a client from this PR can connect to servers from before this PR and that any combination like this still works.  You'll need to least two interfaces running to test.

File: libraries/avatars/src/AvatarData.cpp [hunk 2]
```diff
@@ -1193,10 +1193,15 @@ void AvatarData::sendIdentityPacket() {
...
+    packetList->write(identityData);
+    nodeList->eachMatchingNode(
+        [&](const SharedNodePointer& node)->bool {
+            return node->getType() == NodeType::AvatarMixer && node->getActiveSocket();
+        },
+        [&](const SharedNodePointer& node) {
+            nodeList->sendPacketList(std::move(packetList), *node);
```
Comment (huffman) [inline libraries/avatars/src/AvatarData.cpp]: This probably won't cause issues right now, but if we have multiple avatar mixers you are going to move `packetList` multiple times.

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7824
**Testing Architecture:** 64-bit PC Desktop & 64-bit MacBook Pro
**Testing OS:** Windows 10 & OS X El Capitan
**Testing Locations:** music, sandbox
**Additional Test Results/Observations:** Tested on MAC release 4825 and on a Windows machine PR 7824.  Successfully connected both release and PR builds with each other.  Screen shots included.
**Successful Logout/Crash:** Successful logout.
music test pc pr7824
sandbox test pc pr7824
music test mac 4825
screen shot 2016-05-06 at 1 23 16 pm

**Pull request merged**